src/server.rs

```diff
@@ -83,22 +82,87 @@ impl<'f> FromForm<'f> for Options {
     }
 }
 
+impl Options {
+    pub fn from_rawstr(string: &str) -> Result<Options, HandlerError> {
+        let items = FormItems::from(string);
+        let mut opt = Options {
+            index: None,
+            limit: None,
+            status: None,
+        };
+
+        for item in items {
+            match item.key.to_lowercase().as_ref() {
```

Comment:
I'll note a minor difference between this and the previous FromForm: item.value is no longer url_decoded, but I don't think it matters

Reply:
Ah! Good catch! Forgot to add the `.map()`  call to status. And I think I know why the compiler was complaining about the f() in `map_err` so I can tighten these up a bit.